app/controllers/miq_ae_tools_controller.rb

```diff
@@ -127,9 +127,13 @@ def cancel_import
   end
 
   def import_via_git
-    git_based_domain_import_service.import(params[:git_repo_id], params[:git_branch_or_tag], current_tenant.id)
+    begin
+      git_based_domain_import_service.import(params[:git_repo_id], params[:git_branch_or_tag], current_tenant.id)
 
-    add_flash(_("Imported from git"), :info)
+      add_flash(_("Imported from git"), :info)
+    rescue MiqException::Error
+      add_flash(_("The selected branch or tag does not have valid domain object data"), :error)
```

Comment:
@eclarizio 
https://github.com/ManageIQ/manageiq/blob/master/app/controllers/miq_ae_tools_controller.rb#L239

seems to use the err.message to grab the actual error message is that not needed